### PR TITLE
Revert breaking pipeline changes

### PR DESCRIPTION
**Pull request by qwerty287**
Revert #2180 and #2480 so we can release v2.0 without breaking changes in pipeline config.

After merging #2476 we should apply these changes to a new v2.

After merging this, we should be ready for the 2.0 release.

### CODE REVIEW

**Comment by pat-s** [comment]
> After merging https://github.com/woodpecker-ci/woodpecker/pull/2476 we should apply these changes to a new v2.

You mean v3?

---

I would also be ok with waiting a few days if you think #2476 can make it in and we're good to go then. I can't judge its status...

**Review by 6543 — CHANGES_REQUESTED** [review]
I'm against it ... as it was deprecated last release already ...

**Comment by anbraten** [comment]
> Yes, and it's still deprecated (olds are used only as fallback). However, I'd make this only breaking in a new pipeline config version after we merged https://github.com/woodpecker-ci/woodpecker/pull/2476 .

I am also for leaving the fallback in for now. Biggest issue as mentioned before is that server admins are not enforced to tell their users to update their configs and users might not read migration notes. That's why I would like to get #1572 in to show warnings to the user if the pipeline contains deprecated parts.

**Review by anbraten — COMMENTED** [review]
We need to undo the changes in the yaml schema as well

File: pipeline/frontend/yaml/types/workflow.go [hunk 1]
```diff
@@ -36,6 +36,13 @@ type (
 		Cache    base.StringOrSlice `yaml:"cache,omitempty"`
 		Networks WorkflowNetworks   `yaml:"networks,omitempty"`
 		Volumes  WorkflowVolumes    `yaml:"volumes,omitempty"`
+
+		// Deprecated
+		PlatformDontUseIt string `yaml:"platform,omitempty"` // TODO: remove after v1.2.x version
+		// Deprecated
+		BranchesDontUseIt *constraint.List `yaml:"branches,omitempty"` // TODO: remove after v1.1.x version
+		// Deprecated
+		PipelineDontUseIt ContainerList `yaml:"pipeline,omitempty"` // TODO: remove after v1.1.x version
```
Comment (anbraten) [inline pipeline/frontend/yaml/types/workflow.go:45]:
```suggestion
		// Deprecated
		PlatformDontUseIt string `yaml:"platform,omitempty"` // TODO: remove in next major version
		// Deprecated
		BranchesDontUseIt *constraint.List `yaml:"branches,omitempty"` // TODO: remove in next major version
		// Deprecated
		PipelineDontUseIt ContainerList `yaml:"pipeline,omitempty"` // TODO: remove in next major version
```

**Comment by anbraten** [comment]
> @anbraten the schema was never updated so it should still work

Ahh great 😅

**Review by anbraten — APPROVED** [review]

**Review by 6543 — APPROVED** [review]
well I still want to get rid of it

**Comment by anbraten** [comment]
> well I still want to get rid of it

Sure, next release to do so will come 😉

**Pull request merged**